pandas/tests/scalar/timestamp/methods/test_replace.py

```diff
@@ -21,7 +21,7 @@ def test_replace_out_of_pydatetime_bounds(self):
         # GH#50348
         ts = Timestamp("2016-01-01").as_unit("ns")
 
-        msg = "Out of bounds nanosecond timestamp: 99999-01-01 00:00:00"
+        msg = "Out of bounds timestamp: 99999-01-01 00:00:00 with NPY_DATETIMEUNIT 10"
```

Comment:
Not the ideal error message; just done to get the tests pass but I think should map the code back to a unit

Reply:
from pandas._libs.tslibs.dtypes cimport npy_unit_to_abbrev